src/features/groups/components/InviteMemberDialog/InviteMemberDialogContent.tsx

```diff
@@ -30,23 +30,42 @@ export const InviteMemberDialogContent = ({
   groupId,
 }: IInviteMemberDialogContentProps) => {
   const [isLinkButtonClicked, setIsLinkButtonClicked] = useState(false);
+  const [isHash, setHash] = useState('');
 
+  /**
+   * This useEffect is fired when the `isDialogOpen` state is changed(true/false). if true, the setup is fired.
+   * generating the invitation link hash and put it as state `isHash`
+   */
+  useEffect(() => {
+    const getHash = async () => {
+      if (isDialogOpen) {
+        const result = await putGenerateInvitationLinkHash(groupId);
+        if (!result.ok) {
+          alert('Something went wrong. Please try again');
+          return;
+        }
+        setHash(result.value.invitationLinkHash);
+      }
+    };
+    getHash();
+  }, [isDialogOpen, groupId]);
   /**
    * Handle the link copy button click.
-   * If the generatedLink is not valid, do nothing.
    * If success, generated invitation link URL is copied to clipboard and change the text of button to 'Copied!'
+   * *hash is already generated when the dialog open
    */
   const handleLinkCopy = async () => {
     setIsLinkButtonClicked(true);
-    const result = await putGenerateInvitationLinkHash(groupId);
-
-    if (!result.ok) {
-      alert('Something went wrong. Please try again');
-      return;
+    try {
+      return await navigator.clipboard.writeText(`${CLIENT_BASE_URL}/groups/join/${isHash}`);
+    } catch (err) {
+      console.error(
+        'Error occurred while copying the link:',
+        err instanceof Error ? err.message : err,
+      );
+      alert('Error occurred while copying the link. Please try again.');
     }
-
-    const hash = result.value.invitationLinkHash;
-    navigator.clipboard.writeText(`${CLIENT_BASE_URL}/groups/join/${hash}`);
+    setHash('');
```

Comment:
We should probably not set the hash to an empty string. This functionality works without resetting it because it's renewed every time the dialog opens. Since a state change always triggers a component's re-rendering, we should avoid unnecessary state updates.